### PR TITLE
8331591: sun.font.CharSequenceCodePointIterator is buggy and unused

### PR DESCRIPTION
**Pull request by prrace**
Delete an internal class that has been around for 20 years but never been used.
Verified that is true at least as far back as JDK7
All platforms build. Tests pass.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8331591](https://bugs.openjdk.org/browse/JDK-8331591): sun.font.CharSequenceCodePointIterator is buggy and unused (**Bug** - P4)

### Reviewers
 * [Andy Goryachev](https://openjdk.org/census#angorya) (@andy-goryachev-oracle - no project role)
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19072/head:pull/19072` \
`$ git checkout pull/19072`

Update a local copy of the PR: \
`$ git checkout pull/19072` \
`$ git pull https://git.openjdk.org/jdk.git pull/19072/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19072`

View PR using the GUI difftool: \
`$ git pr show -t 19072`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19072.diff">https://git.openjdk.org/jdk/pull/19072.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19072#issuecomment-2091805224)

### CODE REVIEW

File: src/java.desktop/share/classes/sun/font/CodePointIterator.java [hunk 1]
```diff
@@ -56,15 +56,13 @@ public static CodePointIterator create(char[] text, int start, int limit) {
...
-        return new CharSequenceCodePointIterator(text);
-    }
-
     public static CodePointIterator create(CharacterIterator iter) {
         return new CharacterIteratorCodePointIterator(iter);
     }
+
 }
 
+
```
Comment (andy-goryachev-oracle) [inline src/java.desktop/share/classes/sun/font/CodePointIterator.java:65]: looks like a couple of unnecessary newlines were added inadvertently

**Review by andy-goryachev-oracle — APPROVED** [review]
looks good!

**Review by honkar-jdk — APPROVED** [review]
Cleanup looks good. No other usages of CharSequenceCodePointIterator except in this file. Build works fine too.
Minor: Copyright year needs to be updated.

**Review by honkar-jdk — APPROVED** [review]

File: src/java.desktop/share/classes/sun/font/CodePointIterator.java (hunk 1 above)
Comment (prrace) [reply src/java.desktop/share/classes/sun/font/CodePointIterator.java]: I had already noticed that and removed them

**Pull request closed without merging**